src/VisualStudio/TestUtilities/InProcess/Editor_InProc.cs

```diff
@@ -0,0 +1,254 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Roslyn.VisualStudio.Test.Utilities.Common;
+
+namespace Roslyn.VisualStudio.Test.Utilities.InProcess
+{
+    public class Editor_InProc : InProcComponent
```

Comment:
I'll add comments.

I should mention that I wasn't very clear about why instance methods are required. If they are static methods, they will not be remoted and will not be executed in the host process. We have to call through instance methods to get that to happen. That's why all static methods here are private.